galley/pkg/config/analysis/diag/level.go

```diff
@@ -68,3 +69,36 @@ func GetUppercaseStringToLevelMap() map[string]Level {
 	}
 	return m
 }
+
+// MessageThreshold is a wrapper around Level to use as cobra command line arguments
+type MessageThreshold struct {
+	Level
+}
+
+// String satisfies interface pflag.Value
+func (m *MessageThreshold) String() string {
+	return m.Level.String()
+}
+
+// Type satisfies interface pflag.Value
+func (m *MessageThreshold) Type() string {
+	return "Level"
+}
+
+// Set satisfies interface pflag.Value
+func (m *MessageThreshold) Set(s string) error {
+	l, err := levelFromString(s)
+	if err != nil {
+		return err
+	}
+	m.Level = l
+	return nil
+}
+
+func levelFromString(s string) (Level, error) {
+	val, ok := GetUppercaseStringToLevelMap()[strings.ToUpper(s)]
+	if !ok {
+		return Level{}, fmt.Errorf("%q not a valid option, please choose from: %v", s, GetAllLevelStrings())
```

Comment:
Error strings should just state what's wrong, not be worded as if it were always to be printed. See https://github.com/golang/go/wiki/CodeReviewComments#error-strings for some rationale.

In general the top level caller is responsible for formatting the string further as in the link's example: 
```log.Printf("Reading %s: %v", filename, err)```

In this specific instance you can set validation functions with Cobra so you can do validation at the CLI.